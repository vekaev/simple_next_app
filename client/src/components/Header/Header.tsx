import {
  Box,
  Center,
  ColorScheme,
  Group,
  SegmentedControl,
  useMantineColorScheme,
} from '@mantine/core';
import { useHover } from '@mantine/hooks';
import { IconMoon, IconSun } from '@tabler/icons';

const Header = () => {
  const { hovered, ref } = useHover();
  const { colorScheme, toggleColorScheme } = useMantineColorScheme();

  return (
    <Group position="right" my="xs" mr="md" ref={ref} mb={0}>
      {hovered && 'Also depends on computer theme'}
      <SegmentedControl
        value={colorScheme}
        onChange={(value: ColorScheme) => toggleColorScheme(value)}
        data={[
          {
            value: 'light',
            label: (
              <Center>
                <IconSun size={16} stroke={1.5} />
                <Box ml={10}>Light</Box>
              </Center>
            ),
          },
          {
            value: 'dark',
            label: (
              <Center>
                <IconMoon size={16} stroke={1.5} />
                <Box ml={10}>Dark</Box>
              </Center>
            ),
          },
        ]}
      />
    </Group>
  );
};

export default Header;
