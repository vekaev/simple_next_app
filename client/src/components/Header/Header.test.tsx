import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { renderWithTheme } from '@utils/test/renderWithTheme';
import Header from './Header';

describe('Header', () => {
  test('renders correctly', () => {
    const { container } = renderWithTheme(<Header />);
    expect(container).toMatchSnapshot();

    const lightThemeRadio = screen.getByLabelText('Light');
    const darkThemeRadio = screen.getByLabelText('Dark');

    expect(lightThemeRadio).toBeChecked();
    expect(darkThemeRadio).not.toBeChecked();
  });
  test('should show popover on hover', async () => {
    const { container } = renderWithTheme(<Header />);
    const headerWrapper = container.firstChild as Element;
    let popover = screen.queryByText(/also depends on computer theme/i);

    expect(popover).not.toBeInTheDocument();

    userEvent.hover(headerWrapper);

    popover = screen.getByText(/also depends on computer theme/i);

    expect(popover).toBeInTheDocument();

    userEvent.unhover(headerWrapper);

    popover = screen.queryByText(/also depends on computer theme/i);

    expect(popover).not.toBeInTheDocument();
  });
  test.skip('should toggle color scheme on click', () => {
    renderWithTheme(<Header />);

    let [lightThemeRadio, darkThemeRadio] = screen.getAllByRole('radio');

    expect(lightThemeRadio).toBeChecked();

    // screen.debug(darkThemeRadio);

    [lightThemeRadio, darkThemeRadio] = screen.getAllByRole('radio');

    userEvent.click(darkThemeRadio);
    expect(darkThemeRadio).toBeChecked();
  });
});
