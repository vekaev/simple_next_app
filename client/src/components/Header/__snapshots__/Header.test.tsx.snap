// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    class="mantine-Group-root mantine-kcnu28"
  >
    <div
      class="mantine-SegmentedControl-root mantine-Group-child mantine-s3no6b"
    >
      <span
        class="mantine-SegmentedControl-active mantine-1a1fl52"
      />
      <div
        class="__mantine-ref-control mantine-SegmentedControl-control mantine-SegmentedControl-controlActive mantine-9jajho"
      >
        <input
          checked=""
          class="mantine-1islhj6 mantine-SegmentedControl-input"
          id="mantine-r0-light"
          name="mantine-r0"
          type="radio"
          value="light"
        />
        <label
          class="__mantine-ref-label mantine-SegmentedControl-label mantine-SegmentedControl-labelActive mantine-txut78"
          for="mantine-r0-light"
        >
          <div
            class="mantine-8od8ev"
          >
            <svg
              class="icon icon-tabler icon-tabler-sun"
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <circle
                cx="12"
                cy="12"
                r="4"
              />
              <path
                d="M3 12h1m8 -9v1m8 8h1m-9 8v1m-6.4 -15.4l.7 .7m12.1 -.7l-.7 .7m0 11.4l.7 .7m-12.1 -.7l-.7 .7"
              />
            </svg>
            <div
              class="mantine-1gcamzp"
            >
              Light
            </div>
          </div>
        </label>
      </div>
      <div
        class="__mantine-ref-control mantine-s28z26 mantine-SegmentedControl-control"
      >
        <input
          class="mantine-1islhj6 mantine-SegmentedControl-input"
          id="mantine-r0-dark"
          name="mantine-r0"
          type="radio"
          value="dark"
        />
        <label
          class="__mantine-ref-label mantine-o36ra8 mantine-SegmentedControl-label"
          for="mantine-r0-dark"
        >
          <div
            class="mantine-8od8ev"
          >
            <svg
              class="icon icon-tabler icon-tabler-moon"
              fill="none"
              height="16"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <path
                d="M12 3c.132 0 .263 0 .393 0a7.5 7.5 0 0 0 7.92 12.446a9 9 0 1 1 -8.313 -12.454z"
              />
            </svg>
            <div
              class="mantine-1gcamzp"
            >
              Dark
            </div>
          </div>
        </label>
      </div>
    </div>
  </div>
</div>
`;
